data_model/src/transaction.rs

```diff
@@ -276,12 +275,14 @@ impl Transaction {
     }
 
     /// Adds metadata to the `Transaction`
+    #[must_use]
     pub fn with_metadata(mut self, metadata: UnlimitedMetadata) -> Self {
         self.payload.metadata = metadata;
         self
     }
 
     /// Adds nonce to the `Transaction`
+    #[must_use]
```

Comment:
What about `must_use` for `new()` and/or `sign()` methods? Can't leave comment for its lines